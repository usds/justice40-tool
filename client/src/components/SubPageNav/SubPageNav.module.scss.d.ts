declare namespace SurveyButtonNamespace {
    export interface ISurveyButtonScss {
        subPageNavContainer: string;
    }
  }

declare const SurveyButtonScssModule: SurveyButtonNamespace.ISurveyButtonScss & {
    /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
    locals: SurveyButtonNamespace.ISurveyButtonScss;
  };

  export = SurveyButtonScssModule;
