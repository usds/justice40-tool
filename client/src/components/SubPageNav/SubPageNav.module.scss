@use '../../styles/design-system.scss' as *;

.subPageNavContainer {
    @include u-margin-top(2);
}