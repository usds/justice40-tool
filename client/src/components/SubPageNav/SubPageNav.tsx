import React from 'react';
import {SideNav} from '@trussworks/react-uswds';
import {useIntl} from 'gatsby-plugin-intl';

import LinkTypeWrapper from '../LinkTypeWrapper';
import {PAGES_ENDPOINTS} from '../../data/constants';
import * as COMMON_COPY from '../../data/copy/common';

import * as styles from './SubPageNav.module.scss';

// This prop is used to set which sub-page navigation is active
interface ISubPageNav {
    activeSubPageIndex?: number;
    endPoints: string[];
}
/**
 * This function will take in an endpoint (last segment of the app's URL) and return the appropriate
 * string to find the i18n label
 *
 * @param {string} endPoint
 * @return {string}
 */
const getPageConstant = (endPoint:string) => {
  const intl = useIntl();

  if (endPoint === PAGES_ENDPOINTS.EXPLORE) {
    return intl.formatMessage(COMMON_COPY.HEADER.EXPLORE);
  } else if (endPoint === PAGES_ENDPOINTS.METHODOLOGY) {
    return intl.formatMessage(COMMON_COPY.HEADER.METHODOLOGY);
  } else if (endPoint == PAGES_ENDPOINTS.DOWNLOADS) {
    return intl.formatMessage(COMMON_COPY.HEADER.DOWNLOADS);
  } else if (endPoint == PAGES_ENDPOINTS.TSD) {
    return intl.formatMessage(COMMON_COPY.HEADER.TSD);
  } else if (endPoint == PAGES_ENDPOINTS.ABOUT) {
    return intl.formatMessage(COMMON_COPY.HEADER.ABOUT);
  } else if (endPoint === PAGES_ENDPOINTS.FAQS) {
    return intl.formatMessage(COMMON_COPY.HEADER.FAQS);
  } else if (endPoint == PAGES_ENDPOINTS.PUBLIC_ENG) {
    return intl.formatMessage(COMMON_COPY.HEADER.PUBLIC_ENG);
  } else if (endPoint == PAGES_ENDPOINTS.CONTACT) {
    return intl.formatMessage(COMMON_COPY.HEADER.CONTACT);
  } else if (endPoint == PAGES_ENDPOINTS.PREVIOUS_VERSIONS) {
    return intl.formatMessage(COMMON_COPY.HEADER.PREVIOUS_VERSIONS);
  };
};
/**
 * The main component. It will take in an array of endpoints that the sub-nan should navigate to. It will
 * also take an optional prop that will determine which sub-nav link should be highlighted when the page
 * loads. If the activeSubPageIndex is not provided, it will default to the first index to highlight the
 * first endpoint in the array.
 *
 * @param {number} activeSubPageIndex
 * @param {string[]} endPoints
 * @return {JSX}
 */
const SubPageNav = ({activeSubPageIndex = 0, endPoints}:ISubPageNav) => {
  const subPages = endPoints.map((endPoint, index) =>
    <LinkTypeWrapper
      key={index}
      className={activeSubPageIndex === index ? 'usa-current' : ''}
      url={endPoint}
      internal={true}
      linkText={getPageConstant(endPoint)}
    />,
  );

  // const subPages = [
  //   <LinkTypeWrapper
  //     key={0}
  //     className={activeSubPageIndex === 1 ? 'usa-current' : ''}
  //     url={PAGES_ENDPOINTS.METHODOLOGY}
  //     internal={true}
  //     linkText={intl.formatMessage(COMMON_COPY.HEADER.METHODOLOGY)}
  //   />,
  //   <LinkTypeWrapper
  //     key={1}
  //     className={activeSubPageIndex === 2 ? 'usa-current' : ''}
  //     url={PAGES_ENDPOINTS.DOWNLOADS}
  //     internal={true}
  //     linkText={intl.formatMessage(COMMON_COPY.HEADER.DOWNLOADS)}
  //   />,
  // ];

  return (
    <div className={styles.subPageNavContainer}>
      <SideNav items={subPages}/>
    </div>
  );
};

export default SubPageNav;
