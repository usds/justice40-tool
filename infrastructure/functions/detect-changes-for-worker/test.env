REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1