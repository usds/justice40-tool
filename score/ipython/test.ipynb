{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
