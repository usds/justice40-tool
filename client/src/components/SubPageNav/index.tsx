import SubPageNav from './SubPageNav';

export default SubPageNav;
