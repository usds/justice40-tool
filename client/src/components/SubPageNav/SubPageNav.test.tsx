import * as React from 'react';
import {render} from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import SubPageNav from './SubPageNav';

describe('rendering of the SubPageNav', () => {
  it('checks SubPageNav renders', () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <SubPageNav endPoints={[
            '/methodology',
            '/download',
          ]}/>
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
