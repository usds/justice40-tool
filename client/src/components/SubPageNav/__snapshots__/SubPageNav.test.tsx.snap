// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of the SubPageNav checks SubPageNav renders 1`] = `
<DocumentFragment>
  <div>
    <ul
      class="usa-sidenav"
      data-testid="sidenav"
    >
      <li
        class="usa-sidenav__item"
      >
        <a
          class="usa-link usa-current"
          href="/en/methodology"
        >
          Methodology & data
        </a>
      </li>
      <li
        class="usa-sidenav__item"
      >
        <a
          class="usa-link"
          href="/en/download"
        />
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
